sympy/integrals/tests/test_integrals.py

```diff
@@ -20,6 +21,42 @@
 n = Symbol('n', integer=True)
 f = Function('f')
 
+def test_principal_value():
+
+    g = 1 / x
+    assert Integral(g, (x, -oo, oo)).principal_value() == nan
```

Comment:
@jksuom 
it is giving `nan` because of the reason which I have explained in the documentation's example ( line no. 1268-1274 )

this is the explaination :-
```
This happens so because, the method principal_value returns -oo for Integral from -oo to 0
and +oo for Integral from 0 to oo. While adding both the Integrals i.e. -oo + oo we do not
get 0 (as we expect) but nan since sympy realises that -oo and +oo aren't actual numbers that
can be added/subtracted.
```
does this make sense?

Reply:
The principal value of  this integral is finite. The code should be written to find that value.

Reply:
@jksuom I'm sorry, I didn't get your point here. 
>The principal value of this integral is finite

actually, it is `-oo` from `-oo to 0` and `+oo` from `0 to +oo`. so the resultant is `nan` since sympy realises that -oo and +oo aren't actual numbers that can be added/subtracted.

What exactly is the problem you are pointing out here? Can you please explain?

Reply:
@jksuom this is what I am trying to say
```
>>> from sympy import symbols, integrate, oo
>>> x = symbols('x')
>>> g = 1/x
>>> integrate(g, (x, -oo, 0))
-oo
>>> integrate(g, (x, 0, oo))
oo
>>> integrate(g, (x, -oo, 0)) + integrate(g, (x, 0, oo))
nan
```

Can you please explain what is the problem you are pointing out here?

Reply:
The principal value is not the sum of the two (infinite) integrals from `-oo` to `0` and from `0` to `oo` (which is `nan`). It is the limit of the sum of two finite integrals over suitable subintervals. The limit is finite.